File: aerox.Runtime.Graphics/SamplerSpec.cs
using TerraFX.Interop.Vulkan;

namespace aerox.Runtime.Graphics;

public struct SamplerSpec
{
    public ImageTiling Tiling;
    public ImageFilter Filter;

    public static implicit operator VkSampler(SamplerSpec spec)
    {
        return SGraphicsModule.Get().GetSampler(spec);
    }
}

File: modules/graphics/lib/aerox/graphics/SamplerSpec.cpp
#include "aerox/graphics/SamplerSpec.hpp"

namespace aerox::graphics
{
    SamplerSpec::SamplerSpec(const vk::Filter& inFilter, const vk::SamplerAddressMode& inTiling)
    {
        filter = inFilter;
        tiling = inTiling;
    }

    std::string SamplerSpec::GetId() const
    {
        return std::to_string(static_cast<int>(filter)) + "_" + std::to_string(static_cast<int>(tiling));
    }
}
